### PR TITLE
Associations setup done

### PR DESCRIPTION
**Pull request by rolandfbk**
Below is the list of the tasks that I implemented:

- I created the Classroom class that will have the has-many/belongs-to relationship with Student
- I created the Book and Rental classes
- I create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

** Important Note **
I added the main.rb file to test the code. Some of the variables on that file were to be renamed by the rubocop linter check

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+  attr_reader :rentals, :id
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
     super()
+    @rentals = []
   end
```
Comment (gemmen29) [inline person.rb:14]: - [ ] Kindly create a method that creates a new rental object of class Rental and passes `date`, `book`, and `person` itself as a parameter to achieve the `has-many` association. You may learn more about both ways association from [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (gemmen29) [inline book.rb:9]: - [ ] Kindly create a method that creates a new rental object of class Rental and passes `date`, `person`, and `book` itself as a parameter to achieve the `has-many` association. You may learn more about both ways association from [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @rolandfbk, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
-capitalized_person = CapitalizeDecorator.new(person)
-p capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-p capitalized_trimmed_person.correct_name
+# person = Person.new(22, 'maximilianus')
+# p person.correct_name
+# capitalized_person = CapitalizeDecorator.new(person)
+# p capitalized_person.correct_name
+# capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+# p capitalized_trimmed_person.correct_name
```
Comment (RolandM99) [inline main.rb:12]: - [ ] [**OPTIONAL**] Kindly, remove the commented code from your codebase to make your code clean.

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @rolandfbk, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**